From 10e1f97240ccae3777282280b7135ed76e4f4dc3 Mon Sep 17 00:00:00 2001
From: Ning Sun <ning.sun@intel.com>
Date: Wed, 24 Feb 2016 16:52:23 -0800
Subject: grub2, xen: insert just one dummy command line argument

Generated command lines may repeat the image name, or use
"placeholder", but not both.

Signed-off-by Martin Wilck <martin.wilck@ts.fujitsu.com>
Signed-off-by Ning Sun <ning.sun@intel.com>
---
 tboot/20_linux_xen_tboot |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tboot/20_linux_xen_tboot b/tboot/20_linux_xen_tboot
index 3afcdac..94d2b8e 100644
--- a/tboot/20_linux_xen_tboot
+++ b/tboot/20_linux_xen_tboot
@@ -104,9 +104,9 @@ linux_entry ()
 	echo	'$tmessage'
 	multiboot	${rel_tboot_dirname}/${tboot_basename} ${rel_tboot_dirname}/${tboot_basename} ${tboot_args}
 	echo	'$xmessage'
-	module	${rel_xen_dirname}/${xen_basename} ${rel_xen_dirname}/${xen_basename} placeholder ${xen_args} ${iommu_args}
+	module	${rel_xen_dirname}/${xen_basename} placeholder ${xen_args} ${iommu_args}
 	echo	'$lmessage'
-	module	${rel_dirname}/${basename} ${rel_dirname}/${basename} placeholder root=${linux_root_device_thisversion} ro ${args}
+	module	${rel_dirname}/${basename} placeholder root=${linux_root_device_thisversion} ro ${args}
 EOF
   if test -n "${initrd}" ; then
     message="$(gettext_printf "Loading initial ramdisk ...")"
-- 
1.7.1

